test/integration/014_hook_tests/test_model_hooks_bq.py

```diff
@@ -0,0 +1,146 @@
+        DBTIntegrationTest.setUp(self)
+        self.use_profile('bigquery')
+        self.use_default_project()
+        self.run_sql_file("test/integration/014_hook_tests/seed_model_bigquery.sql")
+
+        self.fields = [
+            'state',
+            'target_name',
+            'target_schema',
+            'target_threads',
+            'target_type',
+            'run_started_at',
+            'invocation_id'
+        ]
+
+    @property
+    def schema(self):
+        return "model_hooks_014"
+
+    @property
+    def profile_config(self):
+        profile = self.bigquery_profile()
+        profile['test']['outputs']['default2']['threads'] = 3
+        return profile
+
+    @property
+    def project_config(self):
+        return {
+            'macro-paths': ['test/integration/014_hook_tests/macros'],
+            'models': {
+                'test': {
+                    'pre-hook': [MODEL_PRE_HOOK],
+
+                    'post-hook':[MODEL_POST_HOOK]
+                }
+            }
+        }
+
+    @property
+    def models(self):
+        return "test/integration/014_hook_tests/models"
+
+    def get_ctx_vars(self, state):
+        field_list = ", ".join(self.fields)
+        query = "select {field_list} from `{schema}.on_model_hook` where state = '{state}'".format(field_list=field_list, schema=self.unique_schema(), state=state)
+
+        vals = self.run_sql(query, fetch='all')
+        self.assertFalse(len(vals) == 0, 'nothing inserted into hooks table')
+        self.assertFalse(len(vals) > 1, 'too many rows in hooks table')
+        ctx = dict(zip(self.fields, vals[0]))
+
+        return ctx
+
+    def check_hooks(self, state):
+        ctx = self.get_ctx_vars(state)
+        print('ctx={}'.format(ctx))
+
+        self.assertEqual(ctx['state'], state)
+        self.assertEqual(ctx['target_name'], 'default2')
+        self.assertEqual(ctx['target_schema'], self.unique_schema())
+        self.assertEqual(ctx['target_threads'], 3)
+        self.assertEqual(ctx['target_type'], 'bigquery')
+        self.assertTrue(ctx['run_started_at'] is not None and len(ctx['run_started_at']) > 0, 'run_started_at was not set')
+        self.assertTrue(ctx['invocation_id'] is not None and len(ctx['invocation_id']) > 0, 'invocation_id was not set')
+
+    @attr(type='bigquery')
+    def test_pre_and_post_model_hooks(self):
+        self.run_dbt(['run'])
+
+        self.check_hooks('start')
+        self.check_hooks('end')
+
+
+class TestBigqueryPrePostModelHooksOnSeeds(DBTIntegrationTest):
+    @property
+    def schema(self):
+        return "model_hooks_014"
+
+    @property
+    def models(self):
+        return "test/integration/014_hook_tests/seed-models"
+
+    @property
+    def project_config(self):
+        return {
+            'data-paths': ['test/integration/014_hook_tests/data'],
+            'models': {},
+            'seeds': {
+                'post-hook': [
+                    'alter table {{ this }} add column new_col int',
```

Comment:
does this work? `alter table ... add column` is not supported on BigQuery! I think a better test might just update a column or insert a record

Reply:
It turns out that we weren't loading the seed file on Postgres or BigQuery because the extension is `.sql` instead of `.csv`, so these hooks never run and the tests always pass.

Good stuff.